File: 백준/Bronze/31520. Champernowne Verification/Champernowne Verification.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n=str(int(input()))
if n=='12345678910':
    print(10)
else:
    flag=True
    for i in range(len(n)):
        if str(i+1)!=n[i]:
            flag=False
            break
    print(n[-1] if(flag) else  -1)

File: 백준/Bronze/31520. Champernowne Verification/Champernowne Verification.cc
#include <iostream>
using namespace std;
int main(void) {
    string n;
    bool flag;
    cin >> n;
    if (n == "12345678910")
        cout << 10;
    else {
        flag = true;
        for (int i = 0; i < n.size(); i++) {
            if (i + 1 != n[i] - 48) {
                flag = false;
                break;
            }
        }
        cout << (flag ? n[n.size() - 1] - 48 : -1);
    }
    return 0;
}